### PR TITLE
Make Prow config more easily configurable from CLI

### PR DESCRIPTION
**Pull request by adrcunha**

### CODE REVIEW

File: ci/prow/Makefile [hunk 1]
```diff
@@ -12,14 +12,26 @@
...
-
-PROW_DIR := $(dir $(abspath $(lastword $(MAKEFILE_LIST))))
-TESTGRID_DIR := $(PROW_DIR)/../testgrid
-SET_CONTEXT := gcloud container clusters get-credentials "$(CLUSTER)" --project="$(PROJECT)" --zone="$(ZONE)"
+CLUSTER            ?= prow
+PROJECT            ?= knative-tests
+ZONE               ?= us-central1-f
+JOB_NAMESPACE      ?= test-pods
+KNATIVE_CONFIG     ?= config_knative.yaml
+SKIP_CONFIG_BACKUP ?=
```
Comment (chaodaiG) [inline ci/prow/Makefile]: what's the use case of skipping config backup? can't we pass in `PROW_CONFIG_GCS` if the concern is to avoid collision?

Comment (adrcunha) [reply ci/prow/Makefile]: I just want to have as many knobs as possible. Sample use case: a staging/temp cluster where I don't care about preserving the config.

Comment (chaodaiG) [reply ci/prow/Makefile]: sgtm

**Pull request merged**